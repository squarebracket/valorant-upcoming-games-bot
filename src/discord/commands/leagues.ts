import { ChatInputCommandInteraction, SlashCommandBuilder } from "discord.js";
import { outputLeagues } from "./lib/output_leagues.ts";

export const data = new SlashCommandBuilder()
  .setName('leagues')
  .setDescription('Returns leagues')

export const execute = async function (interaction: ChatInputCommandInteraction) {
  if (!interaction.guildId) {
    await interaction.reply({ content: `error: no guild id...?` });
  }
  await interaction.deferReply({ ephemeral: true });
  // const leagues = await outputLeagues(interaction);
  // await interaction.followUp({ content: leagues, ephemeral: true });
  await interaction.followUp({ content: await outputLeagues(interaction.guildId!), ephemeral: true });
}
