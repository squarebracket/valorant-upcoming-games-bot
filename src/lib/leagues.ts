import { Config } from "./storage.ts";
import { doRequest } from "./utils.ts";
import { getClient } from "./client.ts";
import { GuildEmoji } from "discord.js";

type Tournament = {
  season: {
    name: string;
  }
}

export type League = {
  id: string;
  name: string;
  slug: string;
  region?:
  "BRAZIL" |
  "CHINA" |
  "EMEA" |
  "INTERNATIONAL" |
  "JAPAN" |
  "KOREA" |
  "LATIN AMERICA" |
  "LATIN AMERICA NORTH" |
  "LATIN AMERICA SOUTH" |
  "NORTH AMERICA" |
  "OCEANIA" |
  "SOUTHEAST ASIA" |
  "VIETNAM";
  tournaments: Tournament[];
  image?: string;
  emoji?: GuildEmoji | string;
}

// note: china has region === 'china' but the other ILs
// have region === 'INTERNATIONAL'
export const defaultFilter = (league: League): boolean => {
  return league.name.includes('Game Changers') || (league.region === 'INTERNATIONAL' && league.name !== 'Challengers South Asia')
    || league.region === 'NORTH AMERICA' || league.region === 'CHINA';
};

let _leagues: League[] = [];
export async function getLeagues(): Promise<League[]> {
  if (_leagues.length) {
    return _leagues;
  } else {
    const client = getClient();

    _leagues = (await doRequest(new URL('https://esports-api.service.valorantesports.com/persisted/val/getLeaguesForStandings?hl=en-US&sport=val')))
      .data.leagues
      .filter((league: League) => !league.name.includes('Last Chance Qualifier') && !league.name.includes('LOCK//IN') && !league.name.includes('Regional League CIS'))
      // .filter((league: League) => parseInt(league.tournaments[0].season.name) >= new Date().getFullYear())
      .map((league: League) => {
        let emoji: GuildEmoji | string | undefined = client.emojis.cache.find(e => e.name === league.slug);
        if (!emoji && /game_changers/.test(league.slug)) {
          emoji = client.emojis.cache.find(e => e.name === 'game_changers');
        } else if (!emoji && (/challengers_kr/.test(league.slug))) {
          emoji = ':flag_kr:';
        } else if (!emoji && (/challengers_jpn/.test(league.slug))) {
          emoji = ':flag_jp:';
        } else if (!emoji && (/challengers_sea_([a-z][a-z])/.test(league.slug))) {
          emoji = `:flag_${RegExp.$1}:`;
        } else if (!emoji && (/challengers|ascension/.test(league.slug) || league.slug === 'valorant_oceania_tour')) {
          emoji = client.emojis.cache.find(e => e.name === 'challengers');
        }
        return {
          id: league.id,
          name: league.name,
          region: league.region,
          slug: league.slug,
          emoji: emoji,
        }
      })

    _leagues.push({
      id: 'a',
      name: 'Game Changers OCE',
      region: 'OCEANIA',
      slug: 'game-changers-oce',
      tournaments: [],
      emoji: client.emojis.cache.find(e => e.name === 'game_changers'),
    });

    return _leagues;
  }
}

export type LeagueFilterFunction = (league: League) => boolean;

export async function filterLeagues(filterFunction: LeagueFilterFunction): Promise<League[]> {
  return (await getLeagues()).filter(filterFunction);
}

export async function getLeagueByName(name: string): Promise<League | undefined> {
  return (await getLeagues()).find((league) => league.name === name);
}

export async function getFilterFuncForObjectIdOrDefault(objectId: string): Promise<LeagueFilterFunction> {
  const config = await Config.findOne({ where: { objectId: objectId } });
  if (config) {
    // @ts-ignore
    const includedLeagues = config.get('mainLeagueIds').concat(config.get('chalLeagueIds'));
    return (league: League) => includedLeagues.includes(league.id);
  } else {
    return defaultFilter;
  }
}

export function sortLeaguesFn(one: League, two: League): number {
  if (/vct|^champions$|masters/i.test(one.name)) {
    return -1;
  } else if (/challengers/i.test(one.name) && /game changers/i.test(two.name)) {
    return -1;
  } else if (/game changers/i.test(one.name) && /challengers/i.test(two.name)) {
    return 1;
  }
  return 0;
}
