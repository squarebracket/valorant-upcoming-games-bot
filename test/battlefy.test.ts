import { describe, expect, test, jest } from '@jest/globals';

import { getLeagueByName } from '../src/lib/leagues.ts';
import { getBattlefy } from '../src/modules/battlefy.ts';
import * as utils from '../src/lib/utils.ts';

import { tourneyJson as nagc_gsl_single } from './data/battlefy_nagc_gsl_single.ts';
import leaguesJson from './data/leagues.json';
import { emotes } from './lib/emotes.ts';

jest.mock('../src/lib/client.ts', () => {
  return {
    getClient: () => {
      return {
        emojis: {
          cache: emotes
        }
      }
    }
  }
});

jest.mock('../src/lib/utils.ts', () => {
  return {
    doRequest: (url: URL, options) => {
      if (url.href.includes('getLeaguesForStandings')) {
        return Promise.resolve(leaguesJson);
      } else if (url.href.includes('cloudfront.net/tournaments/66f5d01a70020f0022f63314')) {
        return Promise.resolve(nagc_gsl_single);
      }
    },
  };
});

describe("battlefy parser tests", () => {
  
  test("minimal args", async () => {
    const league = await getLeagueByName("Game Changers NA");
    const matches = await getBattlefy('66f5d01a70020f0022f63314', league);
    expect(matches).toMatchSnapshot();
  })
  
});