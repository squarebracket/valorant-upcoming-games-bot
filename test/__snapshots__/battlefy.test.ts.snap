// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`battlefy parser tests minimal args 1`] = `
[
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "YDZ Black",
      "name": "YDZ Black",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Skyline Dreamers",
      "name": "Skyline Dreamers",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Shopify Rebellion",
      "name": "Shopify Rebellion",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "YDZ Black",
      "name": "YDZ Black",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "YDZ Black",
      "name": "YDZ Black",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Skyline Dreamers",
      "name": "Skyline Dreamers",
      "result": {
        "mapWins": 1,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Ominous Minions",
      "name": "Ominous Minions",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "AURA Eclipse",
      "name": "AURA Eclipse",
      "result": {
        "mapWins": 1,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "oomfies",
      "name": "oomfies",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Ominous Minions",
      "name": "Ominous Minions",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Ominous Minions",
      "name": "Ominous Minions",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "AURA Eclipse",
      "name": "AURA Eclipse",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "RITUAL RËAL",
      "name": "RITUAL RËAL",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Passionless Project",
      "name": "Passionless Project",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "mesos millionaires",
      "name": "mesos millionaires",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "RITUAL RËAL",
      "name": "RITUAL RËAL",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "RITUAL RËAL",
      "name": "RITUAL RËAL",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Passionless Project",
      "name": "Passionless Project",
      "result": {
        "mapWins": 1,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "HyperSpeed Red",
      "name": "HyperSpeed Red",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Elite Dynasty GC",
      "name": "Elite Dynasty GC",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Teddy Tactics",
      "name": "Teddy Tactics",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "HyperSpeed Red",
      "name": "HyperSpeed Red",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "HyperSpeed Red",
      "name": "HyperSpeed Red",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Elite Dynasty GC",
      "name": "Elite Dynasty GC",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Erinite GC",
      "name": "Erinite GC",
      "result": {
        "mapWins": 1,
      },
    },
    "teamB": {
      "code": "meowtastic",
      "name": "meowtastic",
      "result": {
        "mapWins": 2,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "TENAX GC",
      "name": "TENAX GC",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "meowtastic",
      "name": "meowtastic",
      "result": {
        "mapWins": 1,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "meowtastic",
      "name": "meowtastic",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Erinite GC",
      "name": "Erinite GC",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "SaD FROST",
      "name": "SaD FROST",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "BBB",
      "name": "BBB",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Day By Day",
      "name": "Day By Day",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Nori X",
      "name": "Nori X",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "SaD FROST",
      "name": "SaD FROST",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Day By Day",
      "name": "Day By Day",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Nori X",
      "name": "Nori X",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "BBB",
      "name": "BBB",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Day By Day",
      "name": "Day By Day",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Nori X",
      "name": "Nori X",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Elation Vortex",
      "name": "Elation Vortex",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Students of the Game",
      "name": "Students of the Game",
      "result": {
        "mapWins": 1,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Catnappers",
      "name": "Catnappers",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Ukiyo GC",
      "name": "Ukiyo GC",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Elation Vortex",
      "name": "Elation Vortex",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Catnappers",
      "name": "Catnappers",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Ukiyo GC",
      "name": "Ukiyo GC",
      "result": {
        "mapWins": 0,
      },
    },
    "teamB": {
      "code": "Students of the Game",
      "name": "Students of the Game",
      "result": {
        "mapWins": 2,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Catnappers",
      "name": "Catnappers",
      "result": {
        "mapWins": 1,
      },
    },
    "teamB": {
      "code": "Students of the Game",
      "name": "Students of the Game",
      "result": {
        "mapWins": 2,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Refrag",
      "name": "Refrag",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "SHU GC",
      "name": "SHU GC",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Give Wingman Her Last Nuclear Device",
      "name": "Give Wingman Her Last Nuclear Device",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Purpose Gaming Sirens",
      "name": "Purpose Gaming Sirens",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Refrag",
      "name": "Refrag",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Give Wingman Her Last Nuclear Device",
      "name": "Give Wingman Her Last Nuclear Device",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Purpose Gaming Sirens",
      "name": "Purpose Gaming Sirens",
      "result": {
        "mapWins": 0,
      },
    },
    "teamB": {
      "code": "SHU GC",
      "name": "SHU GC",
      "result": {
        "mapWins": 2,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Give Wingman Her Last Nuclear Device",
      "name": "Give Wingman Her Last Nuclear Device",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "SHU GC",
      "name": "SHU GC",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Shopify Rebellion",
      "name": "Shopify Rebellion",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Ominous Minions",
      "name": "Ominous Minions",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "TENAX GC",
      "name": "TENAX GC",
      "result": {
        "mapWins": 0,
      },
    },
    "teamB": {
      "code": "HyperSpeed Red",
      "name": "HyperSpeed Red",
      "result": {
        "mapWins": 2,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Refrag",
      "name": "Refrag",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Students of the Game",
      "name": "Students of the Game",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "mesos millionaires",
      "name": "mesos millionaires",
      "result": {
        "mapWins": 0,
      },
    },
    "teamB": {
      "code": "Give Wingman Her Last Nuclear Device",
      "name": "Give Wingman Her Last Nuclear Device",
      "result": {
        "mapWins": 2,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "oomfies",
      "name": "oomfies",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "YDZ Black",
      "name": "YDZ Black",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Elation Vortex",
      "name": "Elation Vortex",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "meowtastic",
      "name": "meowtastic",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "SaD FROST",
      "name": "SaD FROST",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "RITUAL RËAL",
      "name": "RITUAL RËAL",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Teddy Tactics",
      "name": "Teddy Tactics",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Day By Day",
      "name": "Day By Day",
      "result": {
        "mapWins": 1,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Shopify Rebellion",
      "name": "Shopify Rebellion",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "HyperSpeed Red",
      "name": "HyperSpeed Red",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Refrag",
      "name": "Refrag",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Give Wingman Her Last Nuclear Device",
      "name": "Give Wingman Her Last Nuclear Device",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "oomfies",
      "name": "oomfies",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Elation Vortex",
      "name": "Elation Vortex",
      "result": {
        "mapWins": 1,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "SaD FROST",
      "name": "SaD FROST",
      "result": {
        "mapWins": 1,
      },
    },
    "teamB": {
      "code": "Teddy Tactics",
      "name": "Teddy Tactics",
      "result": {
        "mapWins": 2,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "Shopify Rebellion",
      "name": "Shopify Rebellion",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Refrag",
      "name": "Refrag",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "completed",
    "teamA": {
      "code": "oomfies",
      "name": "oomfies",
      "result": {
        "mapWins": 2,
      },
    },
    "teamB": {
      "code": "Teddy Tactics",
      "name": "Teddy Tactics",
      "result": {
        "mapWins": 0,
      },
    },
  },
  {
    "league": {
      "emoji": Emote {
        "name": "game_changers_na",
      },
      "id": "106976737954740691",
      "name": "Game Changers NA",
      "region": "NORTH AMERICA",
      "slug": "game_changers_na",
    },
    "startTime": 2024-10-01T21:00:00.000Z,
    "state": "live",
    "teamA": {
      "code": "Shopify Rebellion",
      "name": "Shopify Rebellion",
      "result": {
        "mapWins": 1,
      },
    },
    "teamB": {
      "code": "oomfies",
      "name": "oomfies",
      "result": {
        "mapWins": 0,
      },
    },
  },
]
`;
